From 5848899423dbbf763997577e476cf630689c59c6 Mon Sep 17 00:00:00 2001
From: Arkadiusz Hiler <ahiler@codeweavers.com>
Date: Tue, 21 Sep 2021 12:47:43 +0300
Subject: [PATCH] wine.inf: Use native xinput1_3 for WRC 8, 9 and 10.

WRC games, when run in a win10 prefix, try to load UWP/xinput1_3.dll
that ships with the games. It seems to be mostly-compatible xinput
implementation built on top of windows.gaming.input.dll.

However if we try to replace it with the builtin the game crashes
(that's the "mostly-compatible" part), so we need this override.

Since our windows.gaming.input.dll implementation is just a stub
controllers other than steering wheels (which are handled through
dinput) do not work.

It's possible to make the controllers work in the meantime by switching
the prefix to win7 and adding `WINEDLLOVERRIDES="xinput1_3=b"`, as on
win7 the game uses the regular xinput and is fine with our builtin.

Includes contributions by Sven Arvidsson.

Wine-Bug: https://bugs.winehq.org/show_bug.cgi?id=51643
Link: https://github.com/ValveSoftware/wine/pull/121
CW-Bug-Id: #19337
---
 loader/wine.inf.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 644a2fcce1d..6bf5d49c730 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -4236,6 +4236,9 @@ HKCU,Software\Wine\AppDefaults\NFS16.exe\DllOverrides,"atiadlxx",,"builtin"
 HKCU,Software\Wine\AppDefaults\ShadowOfWar.exe\DllOverrides,"atiadlxx",,"builtin"
 HKCU,Software\Wine\AppDefaults\ShadowOfWar.exe\DllOverrides,"amd_ags_x64",,""
 HKCU,Software\Wine\AppDefaults\DIRT5.exe\DllOverrides,"atiadlxx",,"builtin"
+HKCU,Software\Wine\AppDefaults\WRC8.exe\DllOverrides,"xinput1_3",,"native"
+HKCU,Software\Wine\AppDefaults\WRC9.exe\DllOverrides,"xinput1_3",,"native"
+HKCU,Software\Wine\AppDefaults\WRC10.exe\DllOverrides,"xinput1_3",,"native"
 ;;Likely want *80 and *90 too, but those require removing Wine's manifest files.
 HKCU,Software\Wine\DllOverrides,"atl100",0x2,"native,builtin"
 HKCU,Software\Wine\DllOverrides,"msvcp100",0x2,"native,builtin"
From 13ac81ffd075d9e92486a395be9cfe822562e718 Mon Sep 17 00:00:00 2001
From: Paul Gofman <pgofman@codeweavers.com>
Date: Wed, 6 Oct 2021 16:48:56 +0300
Subject: [PATCH] wine.inf: HACK: Disable libglesv2 for Paradox launcher.

CW-Bug-ID: #19501
---
 loader/wine.inf.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 1cfe19dace6..07d2a576501 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -4237,6 +4237,7 @@ HKCU,Software\Wine\AppDefaults\DIRT5.exe\DllOverrides,"atiadlxx",,"builtin"
 HKCU,Software\Wine\AppDefaults\WRC8.exe\DllOverrides,"xinput1_3",,"native"
 HKCU,Software\Wine\AppDefaults\WRC9.exe\DllOverrides,"xinput1_3",,"native"
 HKCU,Software\Wine\AppDefaults\WRC10.exe\DllOverrides,"xinput1_3",,"native"
+HKCU,Software\Wine\AppDefaults\Paradox Launcher.exe\DllOverrides,"libglesv2",,"disabled"
 ;;Likely want *80 and *90 too, but those require removing Wine's manifest files.
 HKCU,Software\Wine\DllOverrides,"atl100",0x2,"native,builtin"
 HKCU,Software\Wine\DllOverrides,"msvcp100",0x2,"native,builtin"
From 77fd28cba7764a2dafd1659be87aedb62437e4db Mon Sep 17 00:00:00 2001
From: Arkadiusz Hiler <ahiler@codeweavers.com>
Date: Mon, 11 Oct 2021 19:36:10 +0300
Subject: [PATCH] wine.inf: HACK: Disable winusb for Nickelodeon All-Star
 Brawl.

The DLL is just a stub and the game tries to use it for some more exotic
controller handling. It works without it.

CW-Bug-Id: #19533
---
 loader/wine.inf.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 07d2a576501..900f172b325 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -4238,6 +4238,7 @@ HKCU,Software\Wine\AppDefaults\WRC8.exe\DllOverrides,"xinput1_3",,"native"
 HKCU,Software\Wine\AppDefaults\WRC9.exe\DllOverrides,"xinput1_3",,"native"
 HKCU,Software\Wine\AppDefaults\WRC10.exe\DllOverrides,"xinput1_3",,"native"
 HKCU,Software\Wine\AppDefaults\Paradox Launcher.exe\DllOverrides,"libglesv2",,"disabled"
+HKCU,Software\Wine\AppDefaults\Nickelodeon All-Star Brawl.exe\DllOverrides,"winusb",,"disabled"
 ;;Likely want *80 and *90 too, but those require removing Wine's manifest files.
 HKCU,Software\Wine\DllOverrides,"atl100",0x2,"native,builtin"
 HKCU,Software\Wine\DllOverrides,"msvcp100",0x2,"native,builtin"
From 3e7c7f8f7e5a0945a2ba00afeda51e037758a4ac Mon Sep 17 00:00:00 2001
From: Arkadiusz Hiler <ahiler@codeweavers.com>
Date: Fri, 15 Oct 2021 18:17:27 +0300
Subject: [PATCH] wine.inf: Disable libglesv2 for nw.js.

It's Chromium based and doesn't render well without the override, which
is also common with older CEF versions.

CW-Bug-Id: #19483
---
 loader/wine.inf.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 900f172b325..58bf6d4bbab 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -4239,6 +4239,7 @@ HKCU,Software\Wine\AppDefaults\WRC9.exe\DllOverrides,"xinput1_3",,"native"
 HKCU,Software\Wine\AppDefaults\WRC10.exe\DllOverrides,"xinput1_3",,"native"
 HKCU,Software\Wine\AppDefaults\Paradox Launcher.exe\DllOverrides,"libglesv2",,"disabled"
 HKCU,Software\Wine\AppDefaults\Nickelodeon All-Star Brawl.exe\DllOverrides,"winusb",,"disabled"
+HKCU,Software\Wine\AppDefaults\nw.exe\DllOverrides,"libglesv2",,"disabled"
 ;;Likely want *80 and *90 too, but those require removing Wine's manifest files.
 HKCU,Software\Wine\DllOverrides,"atl100",0x2,"native,builtin"
 HKCU,Software\Wine\DllOverrides,"msvcp100",0x2,"native,builtin"

